### ISSUE TITLE
returning undefined from filter

### ISSUE THREAD

**Issue by MrArnoldPalmer** [SBoudrias/Inquirer.js#862]
Hey there. Initially I was looking for a way to filter out empty values from the answers object. I stumbled upon returning `undefined` from the filter function which results in that questions key/value not being present in answers. This was useful when I was prompting the user for a value that was optional, so I would use a filter function similar to:
```
function filterEmpty(val: string): string | void {
  if (typeof val === 'string' && val.length === 0) {
    return undefined;
  }
  return val;
}
```
However, if the question with this filter function also had a validation function, when the user entered an empty value, an error would be thrown.
```
(node:67414) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'length' of undefined
    at InputPrompt.onError (/private/tmp/jsii-config/node_modules/inquirer/lib/prompts/input.js:95:35)
    at SafeSubscriber._next (/private/tmp/jsii-config/node_modules/rxjs/internal/Observable.js:66:21)
    at SafeSubscriber.__tryOrUnsub (/private/tmp/jsii-config/node_modules/rxjs/internal/Subscriber.js:205:16)
    at SafeSubscriber.next (/private/tmp/jsii-config/node_modules/rxjs/internal/Subscriber.js:143:22)
    at Subscriber._next (/private/tmp/jsii-config/node_modules/rxjs/internal/Subscriber.js:89:26)
    at Subscriber.next (/private/tmp/jsii-config/node_modules/rxjs/internal/Subscriber.js:66:18)
    at TakeUntilSubscriber.Subscriber._next (/private/tmp/jsii-config/node_modules/rxjs/internal/Subscriber.js:89:26)
    at TakeUntilSubscriber.Subscriber.next (/private/tmp/jsii-config/node_modules/rxjs/internal/Subscriber.js:66:18)
    at FilterSubscriber._next (/private/tmp/jsii-config/node_modules/rxjs/internal/operators/filter.js:52:30)
    at FilterSubscriber.Subscriber.next (/private/tmp/jsii-config/node_modules/rxjs/internal/Subscriber.js:66:18)
(node:67414) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
```
I tried mitigating by giving the question a default, or by handling `undefined` inside of validation but it doesn't seem to matter.

For now, I have just manually gone through and removed optional answers that are empty after all questions are asked which is a fine solution. However, I wonder if this is an issue that inquirer should handle. It's useful in some cases to allow empty values, but when they aren't empty run some validation logic on them.

Upon further thought inquirer's job is to return an object representing how a user answered the prompted questions, that includes empty values. The library consumer can choose to do what they want with that. The developer experience however is a little wonky in this case so maybe it should be allowed to return `undefined` from filter and have that forwarded as is to validate.

**Comment by anitagraham** [comment]
The underlying problem is mentioned in #845

**Comment by SBoudrias** [comment]
I pushed a fix with Inquirer@7.0.1. Try it out and let me know if the issue persists.
